wagon-providers/wagon-file/src/main/java/org/apache/maven/wagon/providers/file/WaitingLockableFileWriter.java

```diff
@@ -0,0 +1,301 @@
+package org.apache.maven.wagon.providers.file;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * CHECKSTYLE_OFF: LineLength
+ * Adapted from commons-io LockableFileWriter
+ * https://github.com/apache/commons-io/blob/58b0f795b31482daa6bb5473a8b2c398e029f5fb/src/main/java/org/apache/commons/io/output/LockableFileWriter.java
+ * licenced under apache-2.0
+ *
+ * Aside from some clean-up of not used parts, the main difference is that we can optionally wait for a file lock.
+ *
+ * @author <a href="mailto:erikhakan@gmail.com">Erik Håkansson</a>
+ * CHECKSTYLE_ON: LineLength
+ */
+public class WaitingLockableFileWriter extends Writer
```

Comment:
The `LockableFileWriter` from Commons IO is a good idea, but not really suited for this IMHO, because Wagon transports opaque binary files and not text files what I writer is for. Am I mistaken?